opds.py

```diff
@@ -578,78 +578,61 @@ def detailed_author(cls, contributor):
 class AcquisitionFeed(OPDSFeed):
 
     FACET_REL = "http://opds-spec.org/facet"
-    NO_CACHE = object()
+    # TODO: Is this still used?
     FEED_CACHE_TIME = int(Configuration.get('default_feed_cache_time', 600))
 
     @classmethod
-    def groups(cls, _db, title, url, lane, annotator,
-               cache_type=None, force_refresh=False, facets=None,
-               search_engine=None, search_debug=False
+    def groups(cls, _db, title, url, worklist, annotator,
+               facets=None, max_age=None, 
+               search_engine=None, search_debug=False,
     ):
         """The acquisition feed for 'featured' items from a given lane's
         sublanes, organized into per-lane groups.
 
+        NOTE: If the lane has no sublanes, a grouped feed will
+        probably be unsatisfying. Call page() instead with an
+        appropriate Facets object.
+
         :param facets: A GroupsFacet object.
 
-        :return: CachedFeed (if use_cache is True) or unicode
+        :return: A Unicode string containing a (potentially cached) OPDS feed.
         """
-        works_and_lanes = None
-        if lane.children:
-            # Since this lane has children, it's reasonable to at least
-            # try to create a groups feed for it.
-            if not annotator:
-                annotator = Annotator
-            if callable(annotator):
-                annotator = annotator()
-            cached = None
-            use_cache = cache_type != cls.NO_CACHE
-            facets = facets or FeaturedFacets.default(lane)
-            if use_cache:
-                cache_type = cache_type or CachedFeed.GROUPS_TYPE
-                cached, usable = CachedFeed.fetch(
-                    _db,
-                    lane=lane,
-                    type=cache_type,
-                    facets=facets,
-                    pagination=None,
-                    annotator=annotator,
-                    force_refresh=force_refresh,
-                )
-                if usable:
-                    return cached.content
-            works_and_lanes = lane.groups(
-                _db=_db, facets=facets, search_engine=search_engine,
-                debug=search_debug
-            )
+        annotator = cls._make_annotator(annotator)
+        facets = facets or FeaturedFacets.default(worklist.get_library(_db))
 
-        if not works_and_lanes:
```

Comment:
Removing this code makes it the caller's responsibility to decide whether to call page() or groups(), and I think that's where the responsibility should lie.

Reply:
But doesn't the `refresh` method here call the method to generate a group feed? Or is that different?

Reply:
No, this method defines the refresh method, which calls `_generate_groups`, an internal method.

Reply:
Yeah, this makes sense now. I thought I was looking inside the main `fetch` method but see it's the `groups` method.